service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseRoleService.java

```diff
@@ -73,6 +82,48 @@ private void grantCaseAccess(String caseId, Set<String> users, Set<CaseRole> rol
         }
     }
 
+    public void grantCaseAssignmentToUser(String caseId, String user, Set<CaseRole> roles) {
+        grantCaseAssignmentAccess(caseId, Set.of(user), roles);
+        log.info("User {} granted {} to case {}", user, roles, caseId);
+    }
+
+    public void grantCaseAssignmentToLocalAuthority(String caseId, String localAuthority, Set<CaseRole> roles,
+                                                    Set<String> excludeUsers) {
+        Set<String> localAuthorityUsers = getUsers(caseId, localAuthority, excludeUsers, roles);
+        grantCaseAssignmentAccess(caseId, localAuthorityUsers, roles);
+        log.info("Users {} granted {} to case {}", localAuthorityUsers, roles, caseId);
+    }
+
+    private void grantCaseAssignmentAccess(String caseId, Set<String> users, Set<CaseRole> roles) {
+        final Set<String> usersGrantedAccess = Sets.newConcurrentHashSet();
+
+        try {
+            final String userToken = idam.getAccessToken(userConfig.getUserName(), userConfig.getPassword());
+            final String serviceToken = authTokenGenerator.generate();
+            List<CaseAssignedUserRoleWithOrganisation> caseAssignedRoles = users.stream()
+                .map(user -> CaseAssignedUserRoleWithOrganisation.builder()
+                    .caseDataId(caseId)
+                    .userId(user)
+                    .caseRole(CaseRole.LASOLICITOR.formattedName())
+                    .build())
+                .collect(Collectors.toList());
+
+            addCaseAssignedUserRolesRequest.setCaseAssignedUserRoles(caseAssignedRoles);
```

Comment:
AddCaseAssignedUserRolesRequest addCaseAssignedUserRolesRequest = 
                AddCaseAssignedUserRolesRequest.builder()
                    .caseAssignedUserRoles(caseAssignedRoles)
                    .build();

Reply:
done.. added this